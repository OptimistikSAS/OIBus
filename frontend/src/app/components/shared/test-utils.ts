import { Provider } from '@angular/core';
import { NgbConfig } from '@ng-bootstrap/ng-bootstrap';

const NO_ANIMATION_NGB_CONFIG: NgbConfig = { animation: false };

export const noAnimation: Provider = { provide: NgbConfig, useValue: NO_ANIMATION_NGB_CONFIG };
