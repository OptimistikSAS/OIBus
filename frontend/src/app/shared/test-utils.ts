import { Provider } from '@angular/core';
import { NgbConfig } from '@ng-bootstrap/ng-bootstrap';
import { Page } from './types';

const NO_ANIMATION_NGB_CONFIG: NgbConfig = { animation: false };

export const noAnimation: Provider = { provide: NgbConfig, useValue: NO_ANIMATION_NGB_CONFIG };

export function toPage<T>(content: Array<T>, totalElements: number = content.length, num = 0, size = Math.max(20, totalElements)): Page<T> {
  return {
    content,
    totalElements,
    number: num,
    size,
    totalPages: Math.ceil(totalElements / size)
  };
}

export function emptyPage<T>(size = 20): Page<T> {
  return {
    totalElements: 0,
    content: [] as Array<T>,
    size,
    totalPages: 0,
    number: 0
  };
}
