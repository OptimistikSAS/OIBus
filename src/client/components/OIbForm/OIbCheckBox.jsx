import React from 'react'
import PropTypes from 'prop-types'
import { FormGroup, Label, Input, CustomInput } from 'reactstrap'

const OIbCheckBox = ({ label, value, name, onChange, defaultValue, switchButton }) => {
  // if no value was found, load the context with the default value
  // (this will cause a rerender with the correct value)
  React.useLayoutEffect(() => {
    if (value === null) onChange(name, defaultValue)
  }, [value])
  const handleChange = (event) => {
    const { target } = event
    const { checked } = target
    onChange(name, checked, null)
  }
  if (value === null) return null
  return (
    <FormGroup>
      {switchButton ? (
        <CustomInput
          type="switch"
          id={name}
          name={name}
          label={label}
          onChange={handleChange}
          checked={value}
          color="secondary"
        />
      ) : (
        <>
          <Label>{label}</Label>
          <Input
            className="oi-form-input"
            type="checkbox"
            id={name}
            name={name}
            onChange={handleChange}
            checked={value}
            style={{ position: 'relative', left: '1.5rem' }}
          />
        </>
      )}
    </FormGroup>
  )
}
OIbCheckBox.propTypes = {
  name: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  value: PropTypes.bool,
  defaultValue: PropTypes.bool.isRequired,
  switchButton: PropTypes.bool,
}

OIbCheckBox.defaultProps = { value: null, switchButton: false }

export default OIbCheckBox
