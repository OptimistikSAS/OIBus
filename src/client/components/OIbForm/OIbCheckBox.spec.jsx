import React from 'react'
import ReactDOM from 'react-dom'
import { act } from 'react-dom/test-utils'

import OIbCheckBox from './OIbCheckBox.jsx'

let container
beforeEach(() => {
  container = document.createElement('div')
  document.body.appendChild(container)
})

afterEach(() => {
  document.body.removeChild(container)
  container = null
})

describe('OIbCheckBox', () => {
  test('check CheckBox with value(=true)', () => {
    act(() => {
      ReactDOM.render(<OIbCheckBox
        label="label"
        value
        name="name"
        onChange={() => (1)}
        defaultValue={false}
      />, container)
    })
    expect(container).toMatchSnapshot()
  })
  test('check CheckBox with value(=true) and switch mode', () => {
    act(() => {
      ReactDOM.render(<OIbCheckBox
        label="label"
        value
        name="name"
        onChange={() => (1)}
        defaultValue={false}
        switchButton
      />, container)
    })
    expect(container).toMatchSnapshot()
  })
  test('check CheckBox with value(=false)', () => {
    act(() => {
      ReactDOM.render(<OIbCheckBox
        label="label"
        value={false}
        name="name"
        onChange={() => (1)}
        defaultValue={false}
      />, container)
    })
    expect(container).toMatchSnapshot()
  })
})
