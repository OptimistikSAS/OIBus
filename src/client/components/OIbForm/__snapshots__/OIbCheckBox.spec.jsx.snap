// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OIbCheckBox check CheckBox with value(=false) 1`] = `
<div>
  <div
    class="form-group"
  >
    <label
      class=""
    >
      label
    </label>
    <input
      class="oi-form-input form-check-input"
      id="name"
      name="name"
      style="position: relative; left: 1.5rem;"
      type="checkbox"
    />
  </div>
</div>
`;

exports[`OIbCheckBox check CheckBox with value(=true) 1`] = `
<div>
  <div
    class="form-group"
  >
    <label
      class=""
    >
      label
    </label>
    <input
      checked=""
      class="oi-form-input form-check-input"
      id="name"
      name="name"
      style="position: relative; left: 1.5rem;"
      type="checkbox"
    />
  </div>
</div>
`;

exports[`OIbCheckBox check CheckBox with value(=true) and switch mode 1`] = `
<div>
  <div
    class="form-group"
  >
    <div
      class="custom-switch custom-control"
    >
      <input
        checked=""
        class="custom-control-input"
        color="secondary"
        id="name"
        name="name"
        type="checkbox"
      />
      <label
        class="custom-control-label"
        for="name"
      >
        label
      </label>
    </div>
  </div>
</div>
`;
